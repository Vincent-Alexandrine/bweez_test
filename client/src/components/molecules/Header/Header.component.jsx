import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Styled from './Header.styled.js';

function Header({ getCurrency, ...props }) {
  const [t] = useTranslation();
  return (
    <Styled {...props}>
      <h1>{t('title')}</h1>
      <input type="search" onChange={getCurrency} />
    </Styled>
  );
}

Header.propTypes = {
  getCurrency: PropTypes.func,
};

export default Header;
