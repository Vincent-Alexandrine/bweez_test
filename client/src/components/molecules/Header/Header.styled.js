import styled from 'styled-components';
import media from 'Styles/methods/media';

export default styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;

  ${media.phone`
    flex-direction: column;
    justify-content: space-around;
  `}
  [type="search"] {
    border-width: 0;
    padding: .5rem .5rem .5rem 1rem;
    background: #e6e6e6;
    border-radius: 2.5rem;
    transition: all .1s ease-out;
    border-style: solid;
    &:focus {
      outline: none;
      background: transparent;
      box-shadow: inset 3px -2px 4px #cccccc,
        inset 0px 1px 3px #cccccc,
        inset 3px 1px 4px #cccccc;
    }
  }
`;
