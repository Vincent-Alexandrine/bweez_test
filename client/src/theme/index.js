import ThemeService from './theme.service.js';

export default ThemeService;
