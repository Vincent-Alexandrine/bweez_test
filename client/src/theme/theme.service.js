/*
  getTheme(string) -> theme
  addTheme(string, theme) -> void || true || promise
  removeTheme(string) -> void || true || promise

  validateTheme(theme) -> boolean
*/

const themesMap = new Map();

class ThemeService {
  static instance;

  constructor() {
    if (ThemeService.instance)
      return ThemeService.instance;

    ThemeService.instance = this;
  }

  get(name) {
    return themesMap.get(name);
  }

  add({ name, ...theme }) {
    themesMap.set(name, theme);
  }

  names() {
    return [...themesMap.keys()];
  }

  remove(name) {
    return themesMap.delete(name);
  }
}

const singleton = new ThemeService();

export default singleton;
