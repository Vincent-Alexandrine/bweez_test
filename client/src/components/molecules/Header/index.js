import Header from './Header.component.jsx';

export default Header;
