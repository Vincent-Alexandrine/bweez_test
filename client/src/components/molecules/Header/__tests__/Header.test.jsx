import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import theme from 'Styles/vManuscript';

import Header from '../Header.component.jsx';
import Styled from '../Header.styled.js';

test('Header component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Header styled component should match snapshot', () => {
  const component = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Styled />
      </ThemeProvider>
    );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
