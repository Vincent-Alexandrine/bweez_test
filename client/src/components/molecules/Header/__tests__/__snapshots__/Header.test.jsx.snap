// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 [type="search"] {
  border-width: 0;
  padding: .5rem .5rem .5rem 1rem;
  background: #e6e6e6;
  border-radius: 2.5rem;
  -webkit-transition: all .1s ease-out;
  transition: all .1s ease-out;
  border-style: solid;
}

.c0 [type="search"]:focus {
  outline: none;
  background: transparent;
  box-shadow: inset 3px -2px 4px #cccccc, inset 0px 1px 3px #cccccc, inset 3px 1px 4px #cccccc;
}

@media (max-width:576px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<div
  className="c0"
>
  <h1>
    title
  </h1>
  <input
    type="search"
  />
</div>
`;

exports[`Header styled component should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 [type="search"] {
  border-width: 0;
  padding: .5rem .5rem .5rem 1rem;
  background: #e6e6e6;
  border-radius: 2.5rem;
  -webkit-transition: all .1s ease-out;
  transition: all .1s ease-out;
  border-style: solid;
}

.c0 [type="search"]:focus {
  outline: none;
  background: transparent;
  box-shadow: inset 3px -2px 4px #cccccc, inset 0px 1px 3px #cccccc, inset 3px 1px 4px #cccccc;
}

@media (max-width:576px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<div
  className="c0"
/>
`;
